src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformStringTest.cs

```diff
@@ -0,0 +1,389 @@
+            PlatformSpecificAPI();
+
+        if (OperatingSystem.IsOSPlatform([|""windows8.0""|])) // Version part not allowed: The platform 'windows8.0' is not a known platform name
+            PlatformSpecificAPI();
+
+        Debug.Assert(OperatingSystem.IsOSPlatformVersionAtLeast([|""windows8.0""|], 1, 2, 3, 4));
+        Debug.Assert(OperatingSystem.IsOSPlatform([|""invalid""|]));
+    }
+
+    public void PlatformSpecificAPI() { }
+}";
+            await VerifyAnalyzerAsyncCs(csSource);
+        }
+
+        [Fact]
+        public async Task IsOsPlatformWithInvalidPlatformWarnsVB()
+        {
+            var vbSource = @"
+Imports System
+Imports System.Runtime.Versioning
+
+Public Class Test
+    Public Sub M1()
+        If OperatingSystem.IsOSPlatform([|""x""|]) Then PlatformSpecificAPI()
+        If OperatingSystem.IsOSPlatformVersionAtLeast([|""x""|], 1) Then PlatformSpecificAPI()
+        If Not OperatingSystem.IsOSPlatformVersionAtLeast([|""window""|], 8, 100) Then PlatformSpecificAPI()
+        If OperatingSystem.IsOSPlatform(""windows"") Then PlatformSpecificAPI()
+        If OperatingSystem.IsOSPlatformVersionAtLeast(""ios"", 12, 100, 0, 1) Then PlatformSpecificAPI()
+        If OperatingSystem.IsOSPlatform([|""windows8.0""|]) Then PlatformSpecificAPI()
+    End Sub
+
+    Public Sub PlatformSpecificAPI()
+    End Sub
+End Class";
+            await VerifyAnalyzerAsyncVb(vbSource);
+        }
+
+        [Fact]
+        public async Task MoreInvalidPlatformStringsWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+public class Test
+{
+    [[|SupportedOSPlatform(""Ios-4.1"")|]]
+    public void UnsupportedOSPlatformIosDash4_1() { }
+
+    [[|SupportedOSPlatform(""Ios*4.1"")|]]
+    public void UnsupportedOSPlatformIosStar4_1() { }
+
+    [[|SupportedOSPlatform(null)|]]
+    public void UnsupportedOSPlatformWithNullString() { }
+
+    [[|SupportedOSPlatform(""Linux_4.1"")|]]
+    public void SupportedLinux_41() { }
+
+    [[|UnsupportedOSPlatform("""")|]]
+    public void UnsupportedWithEmptyString() { }
+}";
+            await VerifyAnalyzerAsyncCs(source);
+        }
+
+        private static async Task VerifyAnalyzerAsyncCs(string sourceCode)
+        {
+            var test = PopulateTestCs(sourceCode);
+            await test.RunAsync();
+        }
+
+        private static async Task VerifyAnalyzerAsyncCs(string sourceCode, string editorconfigText)
+        {
+            var test = PopulateTestCs(sourceCode);
+            test.TestState.AdditionalFiles.Add((".editorconfig", editorconfigText));
+            await test.RunAsync();
+        }
+
+        private static VerifyCS.Test PopulateTestCs(string sourceCode, params DiagnosticResult[] expected)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = sourceCode,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                MarkupOptions = MarkupOptions.UseFirstDescriptor,
```

Comment:
Is the other descriptor not tested currently?

Reply:
No, i guess if the analyzer has multiple descriptors it requires to choose which one to use, but for the test scenarios there is no case it could return more than one descriptor

Reply:
> for the test scenarios there is no case it could return more than one descriptor

I mean which tests use which descriptor?

If I understand this property correctly, it means that diagnostics specified as markup (i.e. `[|SomeCode|]`) are expected to have the first descriptor. And because all tests are using it, this means the other descriptor is never encountered in tests.

Correct me if I'm wrong please.

Reply:
Never mind, Just read the documentation and found:

> Use the first ***matching*** diagnostic descriptor when multiple diagnostics match the syntax.

Reply:
> Use the first matching diagnostic descriptor when multiple diagnostics match the syntax.

yes, and i meant there is no test/syntax returning multiple descriptor